torch/nn/_functions/rnn.py

```diff
@@ -344,22 +344,21 @@ def backward_extended(self, grad_output, grad_hy):
 
 def RNN(*args, **kwargs):
 
+    # Hack for the tracer that allows us to represent RNNs as single
+    # nodes and export them to ONNX in this form
+    # Check the first argument explicitly to reduce the overhead of creating
+    # the lambda
+    import torch.onnx.symbolic
+
+    @torch.onnx.symbolic_override_first_arg_based(
+        torch.onnx.symbolic.RNN_symbolic_builder(*args, **kwargs)
```

Comment:
This will execute `RNN_symbolic_builder` no matter if we need it or not, and we don't want to do this unnecessarily

Reply:
It won't even cause additional overhead - it will also throw if it finds features unsupported by ONNX.

Reply:
ok, you're right. I assumed RNN() is invoked only once during model construction (which would make more sense) but apparently we reconstruct it on each forward(): https://github.com/pytorch/pytorch/blob/27d7182d6c7e223e04166f33d5ec46ef8b510944/torch/nn/modules/rnn.py#L191 (btw, why?)

Reply:
Because it's an autograd function, and we need an instance for every application